storage/oasis/client_test.go

```diff
@@ -5,50 +5,82 @@ import (
 	"os"
 	"testing"
 
-	"github.com/oasisprotocol/oasis-indexer/tests"
-	config "github.com/oasisprotocol/oasis-sdk/client-sdk/go/config"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/oasisprotocol/oasis-indexer/config"
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi/damask"
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi/history"
+	"github.com/oasisprotocol/oasis-indexer/tests"
 )
 
 const (
 	pastHeight   = 8048956      // oasis-3 genesis height
 	futureHeight = 100000000000 // an unreasonably high height
 )
 
-func newClientFactory() (*ClientFactory, error) {
-	network := &config.Network{
-		ChainContext: os.Getenv("CI_TEST_CHAIN_CONTEXT"),
-		RPC:          os.Getenv("CI_TEST_NODE_RPC"),
+func init() {
+	history.Chains["ci-test"] = &history.History{
```

Comment:
okay this stuff sucks. so much code relies on internally looking at `history.Chains[aChainNameYouGaveMe]`. but that makes it less nice to use custom chains, e.g. a local testnet. I wonder if we could make it more viable to pass an actual *history.History around and chain names less

Reply:
That sounds reasonable from afar. My guess is it's just the difference between having the chain name + a bunch of utility functions that look at a hardcoded history, or representing those functions as methods of the history object.

Reply:
added an alternative SourceConfig to specify one rpc url and a custom chain context

Reply:
added, use SingleNetwork